### PR TITLE
Separate test code to test-purpose package

### PR DESCRIPTION
**Pull request by FlowingSPDG**
Moved test codes to different package.
There are duplicated codes about helper functions(toInt/toFloat32), so we can make it public or move to helper package imo.

Also, let me know if I can help developing this for CS2.

### CODE REVIEW

**Comment by janstuemmel** [comment]
I want this lib as flat as possible, so maybe just make helpers public. I  would really like to understand why we have to move test files to a seperate package? Is there a sepecial use case for that or just convention?

> Also, let me know if I can help developing this for CS2.

Sure. I really would like to review a few pull requests...

File: cs2log_test.go [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
+	"strconv"
 	"strings"
 	"testing"
 	"time"
+
+	cs2log "github.com/janstuemmel/cs2-log"
 )
 
-func ExampleParse() {
+func exampleParse() {
```
Comment (janstuemmel) [inline cs2log_test.go]: does this show up as an example in godoc?

File: cs2log_test.go [hunk 2]
```diff
@@ -1118,3 +1121,28 @@ func strip(s string) string {
...
+func toFloat32(v string) float32 {
+
+	i, err := strconv.ParseFloat(v, 32)
+
+	if err != nil {
+		return float32(0)
+	}
+
+	return float32(i)
+}
```
Comment (janstuemmel) [inline cs2log_test.go]: we do not need them to be in the test file

File: cs2log_test.go [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
 
 import (
 	"fmt"
 	"regexp"
+	"strconv"
 	"strings"
 	"testing"
 	"time"
+
+	cs2log "github.com/janstuemmel/cs2-log"
```
Comment (janstuemmel) [inline cs2log_test.go:11]: no need for naming here as the packages name is `cs2log`

**Comment by FlowingSPDG** [comment]
Hi, first of all thank you for reviewing this! You're doing amazing job here.

> I want this lib as flat as possible, so maybe just make helpers public. 

Got it, I'll make helper functions public.

> I would really like to understand why we have to move test files to a seperate package? Is there a sepecial use case for that or just convention?

As far as I understand, that makes sure that tests to execute process as same as the other library do because it needs to import package as same way as the others do.
e.g. it's impossible to execute private functions or edit private variables.
Also, few test-purpose variables/functions gets mixed up on production code sometimes.
(I'm still learning English, I'm sorry if I couldn't express what I'm trying to tell!)

File: cs2log_test.go (hunk 3 above)
Comment (FlowingSPDG) [reply cs2log_test.go:11]: Oops sorry, my auto-correct did it wrong.

File: cs2log_test.go (hunk 1 above)
Comment (FlowingSPDG) [reply cs2log_test.go]: I do not think so, maybe we could use [this way](https://github.com/googleapis/google-cloud-go/blob/main/compute/apiv1/doc.go) instead ?
https://pkg.go.dev/cloud.google.com/go/compute@v1.23.3/apiv1

**Comment by janstuemmel** [comment]
This is ok for me...
Then pls remove the helper functions from test file but do not make them public in main file...

File: cs2log_test.go (hunk 2 above)
Comment (FlowingSPDG) [reply cs2log_test.go]: deleted.

**Comment by FlowingSPDG** [comment]
I made some changes, do you want me to change example functions to https://github.com/googleapis/google-cloud-go/blob/main/compute/apiv1/doc.go way?

File: cs2log_test.go (hunk 1 above)
Comment (janstuemmel) [reply cs2log_test.go]: that does not work as test example, the example gets tested while running `go test -v ./...` and the assertion is the comment below the log

Comment (janstuemmel) [reply cs2log_test.go]: 666ed65edaac307a6973d1953b0f9a1dc719ea5f

File: cs2log_test.go (hunk 3 above)
Comment (janstuemmel) [reply cs2log_test.go:11]: the same with my ide, so its ok i guess

**Review by janstuemmel — APPROVED** [review]
lgtm. i fixed the example tests

**Pull request merged**